xradar/util.py

```diff
@@ -522,3 +523,43 @@ def get_sweep_keys(dt):
             pass
 
     return sweep_group_keys
+
+
+def apply_to_sweeps(dtree, func, pass_sweep_name=False, *args, **kwargs):
+    """
+    Applies a given function to all sweeps in the radar volume within a DataTree.
+
+    Parameters
+    ----------
+    dtree : DataTree
+        The DataTree object containing radar data.
+    func : function
+        The function to apply to each sweep.
+    pass_sweep_name : bool, optional
+        Whether to pass the sweep name to the function. Defaults to False.
+    *args : tuple
+        Additional positional arguments to pass to the function.
+    **kwargs : dict
+        Additional keyword arguments to pass to the function.
+
+    Returns
+    -------
+    DataTree
+        The DataTree object after applying the function to all sweeps.
+
+    Raises
+    ------
+    Exception
+        Re-raises any exception that occurs during the function application.
+    """
+    for key in list(dtree.children):
+        if "sweep" in key:
+            try:
+                if pass_sweep_name:
+                    dtree[key] = func(dtree[key], sweep=key, *args, **kwargs)
+                else:
+                    dtree[key] = func(dtree[key], *args, **kwargs)
+            except Exception as e:
+                raise RuntimeError(f"An error occurred while processing {key}: {e}")
```

Comment:
Maybe re raising like this?

```suggestion
            except Exception as e:
                raise RuntimeError(f"An error occurred while processing {key!r}") from e
```